### PR TITLE
Allow OCM addons to setup metrics collection with Prometheus

### PR DESCRIPTION
**Pull request by mprahl**
Related:
https://github.com/stolostron/backlog/issues/23547

### CODE REVIEW

File: manifests/klusterlet/managed/klusterlet-work-clusterrole-execution.yaml [hunk 1]
```diff
@@ -20,3 +20,7 @@ rules:
 - apiGroups: ["rbac.authorization.k8s.io"]
   resources: ["clusterroles", "roles"]
   verbs: ["get", "list", "watch", "create", "update", "patch", "delete", "escalate", "bind"]
+# Allow OCM addons to setup metrics collection with Prometheus
+- apiGroups: ["monitoring.coreos.com"]
+  resources: ["servicemonitors"]
```
Comment (haoqing0110) [inline manifests/klusterlet/managed/klusterlet-work-clusterrole-execution.yaml:27]: `servicemonitors` is a namespace scoped CRD, as we discussed before in [Code refactor to split permission for registration/work](https://docs.google.com/document/d/1SWT4jcBhn6c3dPt2TyNdh4NqfdxSSDiOlSSYgy6QgqM/edit#heading=h.ba0yojrlee5r), this namespace scoped resource should be put this into `Role open-cluster-management:{{ .KlusterletName }}-work:execution` (this Role is not created so far). @qiujian16 any comments for this?

Comment (qiujian16) [reply manifests/klusterlet/managed/klusterlet-work-clusterrole-execution.yaml:27]: I think this is for addon to create their service monitors? It is  ok to put here if we do not have that role yet, but yes we should put it in a role in the future. A TODO here would be helpful.

Comment (mprahl) [reply manifests/klusterlet/managed/klusterlet-work-clusterrole-execution.yaml:27]: That's right. It's for an addon to create a service monitor.

I just added the `TODO` comment as requested.

**Review by haoqing0110 — APPROVED** [review]
/lgtm

**Comment by mprahl** [comment]
> /lgtm

Could you also please add the `/approve` comment?

**Pull request merged**